### DISCUSSION TITLE
Spaceship failing with `This request is forbidden for security reasons - The API key in use does not allow this request`

### DISCUSSION THREAD

**Discussion by simondelphia** [fastlane/fastlane#19427] [Q&A] [answered]
Ran `fastlane spaceship` to enter the Spaceship playground. Then logged in with an apple ID that has App Manager access (to all apps).

Found a build and ran `build.submit_for_testflight_review!`

Got the error:

> Spaceship::AccessForbiddenError: This request is forbidden for security reasons - The API key in use does not allow this request

The [docs](https://github.com/fastlane/fastlane/blob/master/spaceship/docs/Authentication.md) say

> Login with API tokens is not supported yet as API access is not generally available

Why is it talking about an API key in the error? I do have an api key and it only has developer access, but I'm only setting that api key up in a fastlane lane, which I'm not calling here because it's just the Spaceship playground. If it is using that API key somehow, how can I swap it out?

Thanks

**Comment by simondelphia** [comment, marked as answer]
I was able to make progress on this using `upload_to_testflight` as per the suggestion from @lucgrabowski [here](https://github.com/fastlane/fastlane/issues/19747)

So far don't need Spaceship.

**Reply by NareshPopWay** [reply]
hello,  
I try to upload my app on test flight but XCode give's error  when i click on Distribute app , like below image 

<img width="706" alt="Screenshot 2024-06-13 at 6 08 59 PM (1)" src="https://github.com/fastlane/fastlane/assets/140043602/dfb2e035-c9b0-4f5a-bfe3-6b1d946eb301">

and i not able to upload on test flight, and i already upload 6-5 build on test flight but this time i facing this issue
